lint-testdata.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+root_dir=$(pwd)
+
+GOLANGCI_LINT="${root_dir}/bin/golangci-lint"
+go env -w GOFLAGS=-mod=mod
+
+for project in testdata/*; do
+  (cd $project && pwd && $GOLANGCI_LINT run)
+done
```

Comment:
what about we have switch here where we can call both: 
$GOLANGCI_LINT run and $GOLANGCI_LINT run --fix

Reply:
This sounds better. I tried to implement in the latest update. 

Would that look good to you?